sync-core/src/test/java/com/cloudant/sync/datastore/AttachmentTest.java

```diff
@@ -66,6 +62,11 @@ public void setAndGetAttachmentsTest() throws Exception {
             SavedAttachment savedAtt2 = (SavedAttachment) datastore.attachmentsForRevision
                     (newRevision).get(0);
             Assert.assertArrayEquals(expectedSha1, savedAtt2.key);
+
+            //Assert file length and saved attachment length
+            long expectedLength = f.length();
+            Assert.assertEquals(expectedLength,savedAtt.getSize());
+            Assert.assertEquals(expectedLength,savedAtt2.getSize());
```

Comment:
This doesn't test just your new code, and particularly with the old SavedAttachment code, the length returned by getSize() is for the file on disk anyway.

If you want to check your PreparedAttachment code calculates the right length, instantiate a prepared attachment with an UnsavedAttachment and check the file length of the prepared attachment.

```java
File f = TestUtils.loadFixture("fixture/"+attachmentName);
long expectedLength = f.length();
Attachment att = new UnsavedFileAttachment(f, "text/plain");
PreparedAttachment patt = new PreparedAttachment(att, /* blah */);
Assert.assertEquals(expectedLength, patt.getSize());
```

With a similar implementation, you should add a test that you are calculating the SHA1 correctly.